src/language/providers/completionProvider.ts

```diff
@@ -0,0 +1,187 @@
+  documentation?: string,
+  sortText?: string
+): CompletionItem {
+  const item = new CompletionItem(name, kind);
+  item.detail = detail;
+  item.documentation = documentation;
+  item.sortText = sortText;
+  return item;
+}
+
+function getColumnAtributes(column: TableColumn): string {
+  return `Field: ${column.COLUMN_NAME}\n Type: ${column.DATA_TYPE}\n HAS_DEFAULT: ${column.HAS_DEFAULT}\n IS_IDENTITY: ${column.IS_IDENTITY}\n IS_NULLABLE: ${column.IS_NULLABLE}`;
+}
+
+async function getTableItems(
+  schema: string,
+  name: string
+): Promise<CompletionItem[]> {
+  if (!completionItemCache.has(schema + name)) {
+    const items = await Table.getItems(schema, name);
+    const completionItems = items.map((i) =>
+      createCompletionItem(
+        i.COLUMN_NAME,
+        CompletionItemKind.Field,
+        getColumnAtributes(i),
+        `Schema: ${schema}\nTable: ${name}\n`
+      )
+    );
+    completionItemCache.set(schema + name, completionItems);
+  }
+  return completionItemCache.get(schema + name);
+}
+
+async function getObjectCompletions(
+  curSchema: string,
+  sqlTypes: { [index: string]: CompletionType }
+) {
+  const schemaUpdate: boolean = updateCache.delete(curSchema);
+  if (!completionItemCache.has(curSchema) || schemaUpdate) {
+    const promises = Object.entries(sqlTypes).map(async ([_, value]) => {
+      const data = await Database.getObjects(curSchema, value.type);
+      return data.map((table) =>
+        createCompletionItem(
+          table.name,
+          value.icon,
+          `Type: ${value.label}`,
+          `Schema: ${table.schema}`,
+          value.order
+        )
+      );
+    });
+
+    const results = await Promise.all(promises);
+    const list = results.flat();
+    completionItemCache.set(curSchema, list);
+  }
+  return completionItemCache.get(curSchema);
+}
+
+async function getCompletionItemsForTriggerDot(
+  objectRefs,
+  currentStatement,
+  offset
+): Promise<CompletionItem[]> {
+  let list: CompletionItem[] = [];
+  const curRef = currentStatement.getReferenceByOffset(offset);
+  const curSchema = curRef.object.schema;
+
+  let curRefIdentifier: ObjectRef;
+  let isAlias: boolean = false;
+  for (const ref of objectRefs) {
+    if (curSchema === ref.object.schema || curSchema === ref.alias) {
+      curRefIdentifier = ref;
+      // check if ref is an alias
+      if (curRefIdentifier.alias && curRefIdentifier.object.name) {
+        isAlias = true;
+      }
+    }
+  }
+
+  // grab completion items (column names) if alias and name are defined
+  if (isAlias) {
+    const completionItems = await getTableItems(
+      curRefIdentifier.object.schema,
+      curRefIdentifier.object.name
+    );
+    list.push(...completionItems);
+  } else {
+    const objectCompletions = await getObjectCompletions(
+      curSchema,
+      completionTypes
+    );
+    list.push(...objectCompletions);
+  }
+  return list;
+}
+
+async function getCompletionItemsForOtherTriggers(objectRefs) {
```

Comment:
I think the name of this function could be shorter or more precise. Perhaps `getCompletionItemsForRefs`.

Also, the parameter could do with a type.